src/ExpressionEvaluator/Core/Source/ResultProvider/SystemHelpers.cs

```diff
@@ -0,0 +1,45 @@
+﻿// A few dependencies from System.dll, which we want to avoid when compiling against Desktop Framework 2.0.
+namespace System.Linq
+{
+    internal class Dummy { }
+}
+    
+namespace System.Diagnostics
+{
+    internal static class Debug
+    {
+        [Conditional("DEBUG")]
+        public static void Assert(bool condition, string message = null)
+        {
+            if (!condition)
+            {
+                throw new InvalidOperationException(message ?? "Assertion failed");
+            }
+        }
+    }
+}
+
+namespace System.CodeDom.Compiler
+{
+    [AttributeUsage(AttributeTargets.All, Inherited = false, AllowMultiple = false)]
+    internal sealed class GeneratedCodeAttribute : Attribute
```

Comment:
It's needed for our designer generated resource files (but we want to avoid referencing the entirety of System.dll just to make the generate code behind compile).